src/mirador-viewer/index.js

```diff
@@ -122,6 +124,46 @@ windowSettings.manifestId = manifest;
       miradorDownloadPlugin: {
         restrictDownloadOnSizeDefinition: false
       },
+      requests: {
+        preprocessors: [ // Functions that receive HTTP requests and manipulate them (e.g. to add headers)
+
+          (url, options) => {
+            // Get the token from the dsAuthInfo cookie
+            let authToken;
+            if (document.cookie.includes('dsAuthInfo')) {
+              authToken = document.cookie.split('; ')
+                .find(c => c.startsWith('dsAuthInfo='))
+                .split('=')[1].split('%22')[3];
+            }
+            if (url.match(dspaceRestPath) && authToken) {
```

Comment:
is this check enough to assure that we are talking to "our" rest api? I want to double check that using the option to load an external manifest in mirador we could be not fooled to sent our auth token to a path that look like ours "server/iiif" but on an external server.

Reply:
So I'm passing the `environment.rest.baseUrl` as a separate parameter and using that to check the path. I think this eliminates the possibility of being fooled when a manifest from an external server is loaded. I could have pulled the host from the` location.host` but the extra parameter seems easier.  Let me know if this seems like a solution...